specs/date-picker/date-picker.md

```diff
@@ -0,0 +1,285 @@
+>
+
+### Background
+
+This is a community requested component.
+
+### Use Cases
+
+- John is booking a room online and needs to enter a date of arrival. Clicking on the calendar icon in the date field, brings up a calendar where John can select a check-in date.
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+  
+### Features
+
+- **Date validation:** Verifies that the entered string is a valid date.
+
+- **Calendar view:** Opens a calendar view for selecting a date.
+
+- **Time view:** Opens time selection view with select boxes for selecting hour, minute and meridian.
+
+- **Month view:** Opens a month selection view. Used for quickly changing the calendar view or for selecting a month when type="month".
+
+- **Year view:** Opens a year selection view. Used for quickly changing the calendar view, month select view or for selecting a year when type="year".
+
+- **Localization:** Formats that date for the local market. MM--DD-YYYY, DD--MM-YYYY, etc. It supports country, language, numbering type and calendar type.
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes in order to achieve this component's goals?*
+
+- Many different calendar types, even within the same market.
+- Time zone differences. These may be different between client and server.
+
+
+### Prior Art/Examples
+- [W3C](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html)
+- [FluentUI](https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker)
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+- *A designer customizing the component.*
+
+### API
+*Component Name*
+- `fast-date-picker`
+
+*Props/Attrs*
+- `value` - A predefined date or one picked by the user.
+= `locale` - Market and language
+- `placeholder` - Text to appear when a date isn't present.
+- `name` - the name of the control
+- `appearance` - enum
+    - outline (default)
+    - filled
+- `readonly` - If the date is editable.
+- 'disabled` - If the control is disabled.
+- `required` - The date field is required or optional.
+- `min` - Earliest allowed date.
+- `max` - Latest allowed date.cx
+- `allowTextInput` - Whether the date-picker allows input of a date string directly or not.
+- `ariaLabel` - Aria label for screen reader.
+- `type` - Type of picker. Supports all of the HTML types. `date` | `month` | `week` | `time` | `datetime-local`
+- `selected-dates` - Dates to be selected.
+- `disabled-dates` - Dates that cannot be selected.
+
+*Methods*
+
+
+*Events*
+
+- `change` - callback fired when the entered date has changed.
```

Comment:
I'm noticing that the platform emits `change` events "...when an when an alteration to the element's value is *committed by the user.*" (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event). Can we align to that behavior? I think it will make the component easier to use if it models native control behavior where possible.

We could also implement the [input event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event) behavior, which is more like what you are describing here, to get the best of both worlds.

Reply:
I was going to proxy these from the fast-text-field, but I started thinking, these events should just bubble. I shouldn't have to emit anything. I can call this in the spec though.

Reply:
It was bubbling the change event from the text-field component but not when picking. Updated the date-picker and the spec to reflect this.